app/src/main/java/org/jellyfin/androidtv/preference/BasicPreferenceStore.kt

```diff
@@ -0,0 +1,70 @@
+package org.jellyfin.androidtv.preference
+
+abstract class BasicPreferenceStore : IPreferenceStore {
```

Comment:
Merge BasicPreferenceStore and IPreferenceStore (and rename it back to just PreferenceStore, we don't prefix interfaces with `I`)

Reply:
I've renamed `IPreferenceStore` to `PreferenceStore`. Thanks for the heads up

Merging the two is a bit more difficult in an interface we can't annotate methods as `protected` or `private`:
My goal was to keep `getInt` `getBool`...etc. methods protected so you users of this interface are forced to use the `get()` operator directly.
Additionally, we can't mark the enum serialization code private at all so the interface becomes polluted with this too.

Our options are:
- Leave this as a public interface and abstract class
- Merge them together, then convert PreferenceStore from an interface to an abstract class
- Move all the methods into the interface but have the API polluted by "public" private methods (hence I'm not keen)
- Something else? Happy for any ideas here

Reply:
- [x] Merge them together, then convert PreferenceStore from an interface to an abstract class

This one seems like the proper approach.